### ISSUE TITLE
Gedling Borough Council 2024/25 update

### ISSUE THREAD

**Issue by jamesmacwhite** [robbrad/UKBinCollectionData#1075]
### Name of Council

Gedling Borough Council

### Issue Information

Hi, not really an issue, but Gedling Borough Council recently released the new 2024/25 calendars, and they have been published in iCal from: https://github.com/jamesmacwhite/gedling-borough-council-bin-calendars/commit/b62afe11db8e0f4e341631794c86290be36ec3f4 the upstream source.

Just getting feedback for anyone using the Gedling JSON data in Home Assistant. I have UKBCD configured, and it looks OK with automated searching using the identifiers e.g. Wednesday G2. 

@sym0nd0 I know you use the data, so just checking in that everything works and automatically updated for you? There are no major changes to the API side. I did have to adjust the calculation of the revised collection date -> usual collection date, but this is more for the front end of the site and doesn't directly influence the JSON data.

Gedling released the calendars on around 1st December I believe, there was a small gap between then and when I updated them, but it should have auto updated. Unit tests held up well, so that was nice.

Thanks.

### Verification

- [X] I searched for similar issues at https://github.com/robbrad/UKBinCollectionData/issues?q=is:issue and found no duplicates
- [X] I have checked my address/postcode/UPRN works on the council's website
- [X] I have provided a detailed explanation of the issue as well as steps to replicate the issue
- [X] I understand that this project is run by volunteer contributors therefore completion of this issue cannot be guaranteed

**Comment by jamesmacwhite** [comment]
@sym0nd0 Amazing, thanks for the feedback and confirming. Glad it is working for you and has been mostly seamless!

I will look into some form of monitoring, so I know when Gedling change their refuse search data on the calendar cycles. I thought Refuse calendars got released before December and I was checking regularly, but I think they did it either on 1st December or just after.

The garden collections shouldn't change until March 2025 now, we are about to go into the no collection months for January and February after the December collections finish.

I'll figure out if there's a way, I can leverage the existing API to monitor query responses from Gedling's refuse search and detect the PDF URL being different. That is relying on Gedling being consistent on the calendar filename naming. At least for the refuse one's they were.

Interesting note, Gedling have made a small mistake on one of the calendars.

https://apps.gedling.gov.uk/refuse/data/MondayG3-2025.pdf
- Monday 26th May Green/Glass box collection not identified on calendar with red border but is noted as a changed collection

I have emailed, hopefully they'll issue a revised version and just overwrite the existing one. They don't print them anymore anyway. Maybe they'll consider HTML and iCal as the better solution :), we can hope!
